File: Php/app/Chart/ChartDateTimeValueModel.php
namespace app\Chart{

	use DateTime;

	class ChartDateTimeValueModel extends ChartValueModel {

		public function __construct(DateTime $dateTime){
			parent::__construct($this->formatDate($dateTime));
		}

		private function formatDate(DateTime $dateTime){
			$format = 'Date(%1$s, %2$s, %3$s, %4$s, %5$s, %6$s)';
			$year = $dateTime->format('Y');
			$month = $dateTime->format('n') - 1;
			$day = $dateTime->format('j');
			$hour = $dateTime->format('G');
			$minute = intval($dateTime->format('i'));
			$second = intval($dateTime->format('s'));
			return sprintf($format, $year, $month, $day, $hour, $minute, $second);
		}

	}

}

File: Web/Models/ChartModels/ChartDateTimeValueModel.cs
using System;

namespace Web.Models.ChartModels
{
    public class ChartDateTimeValueModel : ChartValueModel
    {
        public ChartDateTimeValueModel(DateTime val)
            : base(FormatDate(val))
        {
        }

        private static string FormatDate(DateTime dateTime)
        {
            var year = dateTime.Year;
            var month = dateTime.Month - 1;
            var day = dateTime.Day;
            var hour = dateTime.Hour;
            var minute = dateTime.Minute;
            var second = dateTime.Second;
            return $"Date({year}, {month}, {day}, {hour}, {minute}, {second})";
        }
    }
}